Tests/FluentAssertions.Specs/Execution/CallerIdentifierSpecs.cs

```diff
@@ -50,6 +51,296 @@ public void When_variable_name_contains_Should_it_should_identify_the_entire_var
+
+            // Act
+            Action act = () => foo.Should().BeNull();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo to be <null>*");
+        }
+
+        [Fact]
+        public void When_variable_is_not_captured_it_should_use_the_variable_name()
+        {
+            // Arrange & Act
+            Action act = () =>
+            {
+                string foo = "bar";
+                foo.Should().BeNull();
+            };
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo to be <null>*");
+        }
+
+        [Fact]
+        public void When_field_is_the_caller_it_should_use_the_field_name()
+        {
+            // Arrange & Act
+            Action act = () =>
+            {
+                var foo = new Foo();
+                foo.Field.Should().BeNull();
+            };
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo.Field to be <null>*");
+        }
+
+        [Fact]
+        public void When_property_is_the_caller_it_should_use_the_property_name()
+        {
+            // Arrange
+            var foo = new Foo();
+
+            // Act
+            Action act = () => foo.Bar.Should().BeNull();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo.Bar to be <null>*");
+        }
+
+        [Fact]
+        public void When_method_name_contains_get_it_should_not_remove_the_prefix()
+        {
+            // Arrange
+            var foo = new Foo();
+
+            // Act
+            Action act = () => foo.get_BarMethod().Should().BeNull();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo.get_BarMethod() to be <null>*");
+        }
+
+        [Fact]
+        public void When_method_contains_parameters_it_should_add_them_to_caller()
+        {
+            // Arrange
+            var foo = new Foo();
+
+            // Act
+            Action act = () => foo.get_BarMethod("test").Should().BeNull();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo.get_BarMethod(\"test\") to be <null>*");
+        }
+
+        [Fact]
+        public void When_the_caller_contains_multiple_members_it_should_include_them_all()
+        {
+            // Arrange
+            string test1 = "test1";
+            var foo = new Foo
+            {
+                Field = "test3"
+            };
+
+            // Act
+            Action act = () => foo.GetFoo(test1).GetFooStatic("test" + 2).GetFoo(foo.Field).Should().BeNull();
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("*Expected foo.GetFoo(test1).GetFooStatic(\"test\"+2).GetFoo(foo.Field) to be <null>*");
```

Comment:
I think it is not that hard. We could look for the last dot which is not part of arguments list. The reason why I didn't do it like you suggest is that I saw the current behaviour better. Just let me know if you think it is worth doing. I can do it right here (guess it is better) or implement it as a separate PR.

Reply:
@dennisdoomen I would be OK with leaving this as is as long as you are - I kind of like the extra context (although if it were too long via chained method calls, it could we unwieldy - then again with EF this could end up showing `.ToList()` which wouldn't be helpful).